atomkraft/test/trace.py

```diff
@@ -0,0 +1,57 @@
+from os import PathLike
+from atomkraft.utils.project import project_root
+import os.path
+from datetime import datetime
+import pytest
+from ..reactor.reactor import get_reactor
+
+
+trace_test_stub = """
+from modelator.pytest.decorators import itf
+
+pytest_plugins = "{0}"
+
+@itf("{1}", keypath="{2}")
+def test_trace():
+    print("Successfully executed trace {1}")
+"""
+
+
+def test_trace(trace: PathLike, reactor: PathLike, keypath: str):
+    """
+    Test blockchain by running one trace
+    """
+
+    if reactor is None:
+        reactor = get_reactor()
+
+    root = project_root()
+    if not root:
+        raise RuntimeError(
+            "could not find Atomkraft project: are you in the right directory?"
+        )
+
+    tests = os.path.join(root, "tests")
+    if not os.path.isdir(tests):
+        raise RuntimeError(
+            "Atomkraft tests directory not found: have you executed `atomkraft init`?"
+        )
+
+    timestamp = datetime.now().isoformat(timespec="milliseconds")
+    test_name = f"test_{str(trace)}_{timestamp}"
+    test_name = (
+        test_name.replace("/", "_")
+        .replace(".", "_")
+        .replace(":", "_")
+        .replace("-", "_")
+    )
```

Comment:
I think you're trying to convert `test_name` to a valid snake case. You can use the `case-converter` package here :slightly_smiling_face:

Reply:
Right :D

Reply:
I've tried it. `snakecase` produces this

```
test_tracesexample0itfjson_2022_07_28t104954601.py
```

instead of this

```
test_traces_example0_itf_json_2022_07_28T10_49_54_601.py
```

The latter I believe is more readable; so let's leave it as is for now.